src/java/picard/analysis/CompareMetrics.java

```diff
@@ -17,13 +17,22 @@
  * Compare two metrics files.
  */
 @CommandLineProgramProperties(
-        usage = CompareMetrics.USAGE,
-        usageShort = CompareMetrics.USAGE,
+        usage = CompareMetrics.USAGE_SUMMARY + CompareMetrics.USAGE_DETAIL,
+        usageShort = CompareMetrics.USAGE_SUMMARY,
         programGroup = Metrics.class
 )
 public class CompareMetrics extends CommandLineProgram {
-
-    static final String USAGE = "Compare two metrics files";
+    static final String USAGE_SUMMARY =  "Compares two metrics files";
+    static final String USAGE_DETAIL = "Compares the headers of the two input metrics files that have the same structure," +
+            " but come from different underlying metric classes.  Outputs can be either equal" +
```

Comment:
Not sure how to answer your first question.  Here is the code:
 @PositionalArguments(minElements = 2, maxElements = 2)
    public List<File> metricsFiles;

    private static final Log log = Log.getInstance(CompareMetrics.class);

    @Override
    protected int doWork() {
        IOUtil.assertFilesAreReadable(metricsFiles);
        final MetricsFile<?, ?> metricsA = new MetricsFile();
        final MetricsFile<?, ?> metricsB = new MetricsFile();
        try {
            metricsA.read(new FileReader(metricsFiles.get(0)));
            metricsB.read(new FileReader(metricsFiles.get(1)));
            final boolean areEqual = metricsA.areMetricsEqual(metricsB) && metricsA.areHistogramsEqual(metricsB);
            final String status = areEqual ? "EQUAL" : "NOT EQUAL";
            log.info("Files " + metricsFiles.get(0) + " and " + metricsFiles.get(1) + "are " + status);
        } catch (final Exception e) {
            throw new PicardException(e.getMessage());
        }
        return 0;
    }

    public static void main(String[] argv) {
        new CompareMetrics().instanceMainWithExit(argv);
    }

As for your second question, not sure who to ask.  Shall I post the question on HipChat/Hellbender?